haskell-src/Concordium/Wasm.hs

```diff
@@ -311,6 +334,21 @@ data WasmModule
     | WasmModuleV1 (WasmModuleV V1)
     deriving (Eq, Show)
 
+-- Custom implementation of ToJSON for WasmModule
+instance AE.ToJSON WasmModule where
+    toJSON (WasmModuleV0 wasmV0) = AE.object ["version" AE..= (0 :: Int), "content" AE..= AE.toJSON wasmV0]
+    toJSON (WasmModuleV1 wasmV1) = AE.object ["version" AE..= (1 :: Int), "content" AE..= AE.toJSON wasmV1]
+
+-- Custom implementation of FromJSON for WasmModule
+instance AE.FromJSON WasmModule where
+    parseJSON = AE.withObject "WasmModule" $ \obj -> do
+        version <- obj AE..: "version"
+        content <- obj AE..: "content"
+        case version :: Int of
+            0 -> WasmModuleV0 <$> AE.parseJSON content
+            1 -> WasmModuleV1 <$> AE.parseJSON content
+            _ -> fail "Invalid version number"
+
```

Comment:
I think the format `{"version"=0, "content"={"source"="..."}}` is a bit redundant, and something like `{"version"=0, "source"="..."}` would be cleaner. Something like:

```suggestion
instance AE.ToJSON WasmModule where
    toJSON = \case
        WasmModuleV0 wm -> AE.toJSON wm
        WasmModuleV1 wm -> AE.toJSON wm

instance AE.FromJSON WasmModule where
    parseJSON = AE.withObject "WasmModule" $ \obj -> do
        version <- obj AE..: "version"
        case wordToWasmVersion version of
            Just V0 -> WasmModuleV0 . WasmModuleV <$> obj AE..: "source"
            Just V1 -> WasmModuleV1 . WasmModuleV <$> obj AE..: "source"
            Nothing -> fail $ "Unsupported Wasm version " ++ show version
```